src/web/mjs/connectors/OnePunchMan.mjs

```diff
@@ -23,22 +22,20 @@ export default class OnePunchMan extends Connector {
     async _getChapters(manga) {
         let request = new Request(manga.id, this.requestOptions);
         let data = await this.fetchDOM(request, '#Chapters_List a');
+
         return data.map(element => {
             return {
-                id: this.getRootRelativeOrAbsoluteLink(element, this.url),
```

Comment:
I would keep it unchanged, it will return a relative path if the the link is on the same origin(this.url), or the absolute path if the link is pointing to a different origin.

Reply:
I see your point, but I'll actually never expect a same hostname. Also for simplicity I prefer it that way (eg. use only one function instead of two `getAbsolutePath` vs `getRootRelativeOrAbsoluteLink`)